lib/membrane/rtp/vad.ex

```diff
@@ -108,8 +111,21 @@ defmodule Membrane.RTP.VAD do
 
   @impl true
   def handle_process(:input, %Membrane.Buffer{} = buffer, _ctx, state) do
-    vad_extension = Enum.find(buffer.metadata.rtp.extensions, &(&1.identifier == 1))
-    <<_v::1, level::7>> = vad_extension.data
+    {extension, buffer} = Header.Extension.pop_identifier(buffer, state.header_extension_id)
+    handle_if_present(buffer, extension, state)
+  end
+
+  defp handle_if_present(buffer, nil, state), do: {{:ok, buffer: {:output, buffer}}, state}
+
+  defp handle_if_present(buffer, extension, state) do
+    <<_v::1, level::7>> = extension.data
+
+    new_extension = %Header.Extension{
```

Comment:
Why do we pop the extension to bring it back again? You can add functions to just look up the extension if you don't intend to pop that.

Reply:
Mostly because you have to put it back with locally-recognized identifier, ex. `:vad` instead of `1`. So even if there will be a function like `Header.Extension.present?`, you will have to check for extension's presence, then pop the extension with integer identifier and put one with locally-recognized one.
So instead of one pass through extension list (by `Header.Extension.pop`) you have one pass for checking for presence and second for popping/updating extension to provide a locally-recognized id.

Reply:
I'm wondering if we will ever need the identifier to be changed to `:vad`. Is any other element supposed to read that (for serialization or something)? I guess it is fine for debugging purposes but I am probably missing the use case for that in the long run.

Reply:
`OutboundPacketTracer` can remap these local identifiers to integer identifiers expected by a stream receiver.
So one peer, say Chrome, can send VAD with `extension_id=1` but we cannot forward this to other peers carelessly, because e.g. Safari client expects VAD under `extension_id=14`.  These extension ids vary among clients, and we are able to remap them to ids expected by clients only when we know who exactly the receiver is, that is not sooner than in `StreamSendBin`.